### ISSUE TITLE
scripts/include-common  [[: not found

### ISSUE THREAD

**Issue by tom-wa** [ElektraInitiative/libelektra#1323]
```
./reformat-source: 12: ./include-common: [[: not found
```
```
 % ../scripts/configure-debian-musl                                                                                                                                             [128]
../scripts/configure-debian-musl: 12: ../scripts/include-common: [[: not found
../scripts/configure-debian-musl: 12: ../scripts/configure-common: [[: not found
CMake Error: The source directory "/home/thomas/Dev/libelektra/build/-KDB_DEFAULT_STORAGE=ini" does not exist.
```

**Comment by markus2330** [comment]
@e1528532 is this a regression from #1295?

**Comment by e1528532** [comment]
Looks like it, i think the issue is that [[ is only available in bash, zsh and similar. I assume you use debian? It looks like debian uses /bin/dash per default which is quite limited compared to bash.

Since its probably better to avoid such bash specific extensions and try to use whatever the original shell specification gives you due to its wider compatibility, does the following patch fix your problem @tom-wa?
https://gist.github.com/e1528532/f73243d211ea5611c279eebe544c9f5b

I've tried it and it works in bash, zsh and dash both on a arch linux and macOS now, so basically as long as the shell supports the sh specification.

**Comment by markus2330** [comment]
> It looks like debian uses /bin/dash per default which is quite limited compared to bash.

In Debian `/bin/sh` is a symlink, in newer installations usually to `dash`. But it could be to another shell. If you state in the shebang `/bin/sh` you claim you exclusively use POSIX shell features. Thus `dash` does not support much more than POSIX, it is a very good test.

> git.github.com

Please always prefer PRs, then the build server can build it and it is much easier for us to apply the patch.

> I've tried it and it works in bash, zsh and dash both on a arch linux and macOS now

Then there no need to let it other people try it, too ;) Thanks for the extensive tests! I am looking forward to the PR.

Can you please also add a unit test of something that includes scripts/include-common to avoid future regressions?

**Comment by e1528532** [comment]
done and closed in #1326

**Comment by markus2330** [comment]
You can also write `close #xxx` in the commits, then the issues will be closed automatically.
